src/annotator/config.js

```diff
@@ -16,6 +16,25 @@ function configFrom(window_) {
       document.querySelector('link[type="application/annotator+html"]').href,
   };
 
+  // Extract the default query from the URL.
+  //
+  // The Chrome extension or proxy may already have provided this config
+  // via a tag injected into the DOM, which avoids the problem where the page's
+  // JS rewrites the URL before Hypothesis loads.
+  //
+  // In environments where the config has not been injected into the DOM,
+  // we try to retrieve it from the URL here.
+  var directLinkedID = annotationQuery.extractAnnotationQuery(window_.location.href);
+  if (directLinkedID) {
+    Object.assign(config, directLinkedID);
+  }
+
+  // Browser extensions should ignore all configuration from host page except
+  // annotationIDs
+  if (config.app.startsWith("chrome-extension://")) {
```

Comment:
In addition to "chrome-extension", we should also check for the "moz-extension" scheme used by Firefox Web Extensions. For Edge extensions, the scheme is "ms-browser-extension". See https://docs.microsoft.com/en-us/microsoft-edge/extensions/api-support/supported-apis